### PR TITLE
feat: add ui-bootstrap endpoint

### PR DESCRIPTION
**Pull request by chriswk**
- Reducing calls needed for frontend to 1 instead of the current 6

fixes: #789

### CODE REVIEW

**Comment by chriswk** [comment]
We should really discuss the types I've added to the services/stores and which ones we're exposing, I've done the minimum to make this work for now, but am prepared to add the extra elbow grease to pave the way for a smoother development experience further down the line.

Also, for now had to add a @ts-ignore annotation to the call to our `req.user` since `express Request` does not expose a user field. I was trying to add a custom-request.d.ts file to add the definition for the user field but couldn't figure out where to put the file to make `tsc` pick it up

File: src/lib/db/context-field-store.ts [hunk 1]
```diff
@@ -19,8 +22,30 @@ const mapRow = row => ({
...
+    createdAt: Date;
+}
+
 class ContextFieldStore {
-    constructor(db, customContextFields, getLogger) {
+    private db: Knex;
+
+    private logger: Logger;
+
+    constructor(db: Knex, customContextFields, getLogger: LogProvider) {
```
Comment (chriswk) [inline src/lib/db/context-field-store.ts:48]: What should be the type of `customContextFields` here?

File: src/lib/db/project-store.ts [hunk 2]
```diff
@@ -1,23 +1,48 @@
...
+
+    private logger: Logger;
+
+    constructor(db: Knex, getLogger: LogProvider) {
         this.db = db;
         this.logger = getLogger('project-store.js');
     }
 
-    fieldToRow(data) {
+    fieldToRow(data): IProjectInsert {
```
Comment (chriswk) [inline src/lib/db/project-store.ts:37]: What's the type of `data`? Bit surprised that typescript didn't warn me that data was untyped here.

File: src/lib/routes/admin-api/bootstrap-controller.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            strategies,
+            projects,
+        ] = await Promise.all(jobs);
+
+        res.json({
+            ...this.config.ui,
+            unleashUrl: this.config.unleashUrl,
+            baseUriPath: this.config.baseUriPath,
+            version: this.config.version,
+            // @ts-ignore
```
Comment (chriswk) [inline src/lib/routes/admin-api/bootstrap-controller.ts:73]: Don't like this, should really figure out where to expand the typing info for express Request.

File: src/lib/db/context-field-store.ts (hunk 1 above)
Comment (ivarconr) [reply src/lib/db/context-field-store.ts:48]: we should just remove these in v4. not needed when we have a proper API.

**Review by ivarconr — APPROVED** [review]

**Pull request merged**